docs/glossary.md

```diff
@@ -0,0 +1,181 @@
+# Glossary
+
+## Anonymous branch
+
+An anonymous branch is a branch that doesn't have any [named branches](#branch)
```

Comment:
> An anonymous branch is a branch

I think we still need to define what the second "branch" means in this context. Since we mean "branch" in a topological sense, maybe:

```
An anonymous branch is a chain of commits that does not have any named branches pointing to it or to any of its descendants...
```

Reply:
Oops, I meant to write something like that but clearly forgot. Thanks.